Programming/movement

RSGameDev opened the pull request:
Here's what i've done up to this evening. 
I tried amending the scripts i worked on so it adopts the preferred code style for this project.

I still would like to make a little bit more alteration to the movement code but I wanted to push an update so you have what  has been worked on currently.

The movement has been focused on. Now there is lerping, a couple of things that worked ok before aren't working as well now. This is mainly player-enemy interaction for example. Sometimes when the player goes into the enemy it stays on the same tile as the enemy, instead of being pushed back. This is to do with the new lerp feature thats in the game.
So that's something we'll have to address at some point.

RSGameDev committed:
Smooth Movement WIP

RSGameDev committed:
Merge remote-tracking branch 'origin/develop' into programming/movement

RSGameDev committed:
Smooth Movement

I still would like to update some stuff with this but for now here's what's made.

KEREMADALI commented on bpdv3/Assets/Scripts/Floor/TileController.cs:
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
     public GameObject player;
-    
 
+    public bool tilesScrolling;
```
IsPlatformMoving is a better name and make it private if it doesn't have to be public, in that case _isPlatformMoving

KEREMADALI commented on bpdv3/Assets/Scripts/Floor/TileController.cs:
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
     Vector3 direction;
     public float distance;
 
+    Vector3 tileMovement;
```
Please specify accessor and rename to _tileMovement

KEREMADALI commented on bpdv3/Assets/Scripts/Floor/TileController.cs:
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
     {
         direction = Vector3.back;
-        distance = direction.magnitude;
+        distance = 1.25f;
```
Please make 1.25f a constant

KEREMADALI commented on bpdv3/Assets/Scripts/Floor/TileController.cs:
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
             foreach (GameObject gameObject in tilesArray)
             {
-                var tileMovement = direction.normalized * (Time.deltaTime * (1.25f / 4f));
+                tileMovement = direction * (Time.deltaTime * (distance / 3f));                
```
Please make 3f a constant

KEREMADALI commented on bpdv3/Assets/Scripts/Lose Condition/BottomRowHazard.cs:
```diff
@@ -15,7 +15,7 @@ private void OnTriggerStay(Collider other)
         {
             other.GetComponent<Player>().PlayerDied();
-            other.GetComponent<Player>().StartPosition();
+            //other.GetComponent<Player>().StartPosition();
```
Delete if unnecessary

KEREMADALI commented on bpdv3/Assets/Scripts/Mechanics/Anchor.cs:
```diff
@@ -7,21 +7,28 @@ public class Anchor : MonoBehaviour
 {
     public GameObject tileObj;
 
+    Vector3 newPos;
```
please specify accessor and rename to _newPosition

KEREMADALI commented on bpdv3/Assets/Scripts/Mechanics/Anchor.cs:
```diff
@@ -7,21 +7,28 @@ public class Anchor : MonoBehaviour
+
+    [SerializeField] Vector3 tilepos;
+
     private void OnTriggerEnter(Collider other)
```
This method does too much job inside please seperate its responsilibities into smaller methods and make easy to read. It is impossible to understand what happens onTrigger

KEREMADALI replied:
Good updates, you can add comments like "resolved" on fixes

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
 // Script attached to the player.
 public class Player : MonoBehaviour
 {
+    #region Private variables
```
Great :)

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
 
-    bool isAlive;
-
+    public bool isAlive = true;
```
PascalCase -> IsAlive

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
+    public GameObject[] spawnPlayerDetect;
+
     public GameUI gameUIScript;
+    public Vector3 startPos;
```
StartPosition

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
+    #endregion
+
+    #region Public variables
+    public GameObject[] spawnPlayerDetect;
```
Please find a more definitive name

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
+    #region Public variables
+    public GameObject[] spawnPlayerDetect;
+
     public GameUI gameUIScript;
```
Please use PascalCase on public variables

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -6,24 +6,37 @@
-    public GameObject[] spawnPlayerDetect;
+    private void Awake()
+    {
+        _playerMovement = GetComponent<PlayerMovement>();
```
Add requiredComponent at the top of class

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -35,12 +48,23 @@ public void StartPosition()
 
     public void PlayerDied()    
```
KillPlayer()

KEREMADALI commented on bpdv3/Assets/Scripts/Player/Player.cs:
```diff
@@ -35,12 +48,23 @@ public void StartPosition()
+        StartCoroutine(EnableMove());
+    }
+
+    IEnumerator EnableMove()
```
please specifty accessor

KEREMADALI commented on bpdv3/Assets/Scripts/Player/PlayerMovement.cs:
```diff
@@ -5,86 +5,136 @@
+        }
+    }
 
+    void Movement()
```
please specift accessor

KEREMADALI commented on bpdv3/Assets/Scripts/Player/PlayerMovement.cs:
```diff
@@ -5,86 +5,136 @@
+        }
+    }
+
+    void CheckBoundaries()
```
Accessor

KEREMADALI replied:
Please remove magic numbers in this method and add comments to this method

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/Enemy.cs:
```diff
@@ -6,52 +6,57 @@
+    #region Public variables
+    public bool isPushBack; 
+    public bool isAlive = true;
+    public bool isNew = true;
```
IsPushBack, IsAlive, IsNew

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/Enemy.cs:
```diff
@@ -73,11 +78,11 @@ private void OnTriggerEnter(Collider other)
 
     private void EnemyDies()
```
KillEnemy();

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/EnemyMovement.cs:
```diff
@@ -6,31 +6,68 @@
     {
-        enemyScript = GetComponent<Enemy>();
-        nextMoveScript = transform.GetChild(2).GetComponent<EnemyNextMove>();
+        _enemy = GetComponent<Enemy>();
```
Use requeiredocmponent

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/EnemyMovement.cs:
```diff
@@ -6,31 +6,68 @@
+    {
+        if (lerp)
+        {
+            Lerp();
```
Make your code readable for anyone not for only coders. Assume that I dont know how to code. Lerp is a mystery for me, please use a more self explanatory method name

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/EnemyMovement.cs:
```diff
@@ -6,31 +6,68 @@
+    private float startTime;
+    private float journeyLength;
+    public float speed = 1.0F;
+    public bool lerp;
```
What is the usage of this boolean, name does not explain

KEREMADALI commented on bpdv3/Assets/Scripts/enemy/EnemyMovement.cs:
```diff
@@ -77,7 +114,7 @@ public void Movement()
         }
 
-        if (enemyScript.currentEnemyType == Enemy.EnemyType.Rook)
+        if (_enemy.currentEnemyType == Enemy.EnemyType.Rook)
```
An enemy cant be both rook and a shroom therfore use switch or if else

KEREMADALI reviewed (changes requested):
Since it is a new convention for you, it is normal to mistake for some of them, please dont be discoured by my comments. As a reminder;

public  -> PascalCase
private -> _camelCase
local variable -> camelCase
Always define accessors both on variables and methods

RSGameDev reviewed (commented):
Hi, I'll push the changes shortly. Will mark for review again.

RSGameDev committed:
code review corrections

RSGameDev reviewed (commented):
Hi, this is a recent push with code review changes based from the last review.
Let me know if it appears better. Thanks Kerem.

RSGameDev committed:
code review correction 2

Somehow i missed some corrections so this includes the remaining corrections from the original review.

KEREMADALI commented on bpdv3/Assets/Scripts/Floor/TileController.cs:
```diff
@@ -5,53 +5,57 @@
     void Update()
     {
-        if (SceneController.Instance.scrollBeatCount >= 6)
+        if (SceneController.Instance.scrollBeatCount >= s_beatToStartScroll)
```
All of code in here seems like they control the same thing u can create a method for this and that method in Update

KEREMADALI commented on bpdv3/Assets/Scripts/Mechanics/Anchor.cs:
```diff
@@ -5,39 +5,61 @@
-            Vector3 newPos = other.GetComponent<Renderer>().bounds.center;
-            transform.parent.position = new Vector3(newPos.x, newPos.y + 1f, newPos.z);
-            transform.parent.SetParent(other.transform);
+            print("fsavda");
```
Please remove

KEREMADALI commented on bpdv3/Assets/Scripts/Player/PlayerMovement.cs:
```diff
@@ -2,89 +2,150 @@
+        transform.position = Vector3.Lerp(transform.position, _destination, fractionOfJourney);
+        if (fractionOfJourney >= 0.1f || !Player.IsPlayerAlive)
+        {
+            IsLerpingStarted = false;
```
If we removede lerp can this bool deleted?

KEREMADALI reviewed (commented):
Good updates well done

RSGameDev commented:
Thanks for the further comments, ill look at them fully over next couple days and will amend the code accordingly from the notes u have left. I'll look to make the shroom movement better working also as at the moment. It's not in the best shape ^^. Is getting there :). Just having a resting day today, not feeling too well after last night lol.

RSGameDev commented:
Corrected changes. i forgot in a recent push but have done it for the push im about to do. Thanks for the review also.
